modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -486,10 +486,9 @@ public ObjectDefinition publishCustomObjectDefinition(
 			throw new ObjectDefinitionStatusException();
 		}
 
-		int count = _objectFieldPersistence.countByObjectDefinitionId(
-			objectDefinition.getObjectDefinitionId());
+		if (_objectFieldPersistence.countByODI_NotS(objectDefinitionId, true) ==
```

Comment:
You can combine this PR with this one that I just reviewed it: https://github.com/liferay-objects/liferay-portal/pull/687